viewing_party/party.py

```diff
@@ -1,23 +1,142 @@
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+    return None
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    updated_data = user_data.copy()
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            updated_data["watched"].append(movie)
+            updated_data["watchlist"].remove(movie)
+
+    return updated_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    total_ratings = 0.0
+    total_movies = 0
+    avg_rating = 0.0
+
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            total_movies += 1
+            total_ratings += movie["rating"]
+        avg_rating = total_ratings/total_movies
+    return avg_rating

```

Comment:
👍  Good work!

Another way to find the total amount of movies is by getting the length of the user's watched list.

`total_movies = len(user_data["watched"])`